internal/cloud/cloud_status_manager.go

```diff
@@ -166,12 +193,10 @@ func (c *CloudStatusManager) OnChange(ctx context.Context, st store.RStore) {
 	// 3. request started in (1) finishes, sets TokenKnownUnregistered = true
 	// we never make a request post-(2), where the token was registered
 	// This is mitigated by - a) the window between (1) and (3) is small, and b) the user can just click refresh again
-	if time.Now().Before(sleepingAfterErrorUntil) ||
-		state.CloudStatus.Username != "" ||
-		state.CloudStatus.TokenKnownUnregistered ||
-		currentlyMakingRequest {
+	allowedToPerformLookup := !time.Now().Before(sleepingAfterErrorUntil) && !currentlyMakingRequest
+
+	if needsLookup && allowedToPerformLookup {
+		go c.CheckStatus(ctx, st, state.CloudAddress, requestKey, false)
```

Comment:
fwiw, i've been moving more of the subscribers to do their work synchronously (not using goroutines at all), and just doing the parallelism more intelligently in store. it makes testing a bit easier.

Reply:
yeah, that sounds good to keep in mind